sympy/combinatorics/fp_groups.py

```diff
@@ -1381,17 +1391,25 @@ def reidemeister_presentation(fp_grp, H, C=None ,rel=False):
     if not C:
         C = coset_enumeration_r(fp_grp, H)
     C.compress(); C.standardize()
-    define_schreier_generators(C, rel=rel)
+    define_schreier_generators(C, homomorphism=homomorphism)
     reidemeister_relators(C)
     gens, rels = C._schreier_generators, C._reidemeister_relators
-    # To-do change scherier generators correspondance to newset of gens.s
     gens, rels = simplify_presentation(gens, rels, change_gens=True)
 
     C.schreier_generators = tuple(gens)
     C.reidemeister_relators = tuple(rels)
-    if rel:
-        C.schreier_gen_elem = tuple(C._schreier_gen_elem)
-        return C.schreier_generators, C.reidemeister_relators, C.schreier_gen_elem
+    if homomorphism:
+        G = C.fp_group
+        if gens:
+            _G = FpGroup(g[0].group, rels)
```

Comment:
This should be the same as what's returned by `fp_grp.subgroup(H)`. Hmm. Perhaps, it would make sense to postpone homomorphism construction until then, and here just return the dictionary `{gen: _schreier_gen_elem[str(gen)]}`. Then you'd need to add a `homomorphism` keyword to subgroups, and also add some examples there (btw, this should be possible to test now; does it return a valid isomorphism in all cases?).